### PR TITLE
Added parameter filterTypes to schemazen script command

### PR DESCRIPTION
**Pull request by marcio-santos-zocdoc**
@Zocdoc/ci-devs

### CODE REVIEW

File: console/Script.cs [hunk 1]
```diff
@@ -1,90 +1,114 @@
...
+            return 0;
+        }
+
+        private List<string> HandleFilteredTypes() {
+            var filteredTypes = FilterTypes == null ? new List<string>() : FilterTypes.Split(',').ToList();
+
+            foreach (var filterType in filteredTypes) {
+                if (!Database.Dirs.Contains(filterType)) {
+                    Log(TraceLevel.Warning, string.Format("{0} is not a valid type.", filterType));
+                    Log(TraceLevel.Warning, string.Format("Valid types: {0}", Database.ValidTypes));
```
Comment (scott-roepnack-zocdoc) [inline console/Script.cs]: This log is inside the foreach loop, seems kinda strange to keep outputting the same thing.  Can it be moved outside the loop?

**Comment by colin-hanson-zocdoc** [comment]
What's up with Database.cs? It can't be displayed in github because of too many changes... What did you change there?

File: console/DbCommand.cs [hunk 2]
```diff
@@ -45,15 +46,17 @@ public abstract class DbCommand : ConsoleCommand {
...
+
 			if (!string.IsNullOrEmpty(ConnectionString)) {
 				if (!string.IsNullOrEmpty(Server) ||
 			        !string.IsNullOrEmpty(DbName) ||
 					!string.IsNullOrEmpty(User) ||
 					!string.IsNullOrEmpty(Pass)) {
 					throw new ConsoleHelpAsException("You must not provide both a connection string and a server/db/user/password");
 				}
-				return new Database {
+				return new Database(filteredTypes) {
```
Comment (colin-hanson-zocdoc) [inline console/DbCommand.cs:59]: Should probably follow the established pattern here and use object initializer rather than constructor

**Comment by colin-hanson-zocdoc** [comment]
This is a weird implementation to me.  Correct me if I'm wrong, but it looks like we actually go through the effort of loading up all the objects, but then in the end only output the ones we care about?  I'd cut it off way earlier and not load any types that we ultimately don't care out.  This will also make it a lot faster.

See this method https://github.com/Zocdoc/schemazen/blob/master/model/Database.cs#L142

**Comment by colin-hanson-zocdoc** [comment]
The reformat makes this super hard to review... I'm all for formatting, but we should do it in seperate PRs from logical changes

File: console/DbCommand.cs (hunk 2 above)
Comment (marcio-santos-zocdoc) [inline console/DbCommand.cs:59]: :+1:

**Pull request closed without merging**